kensu/google/cloud/bigquery/extractor.py

```diff
@@ -49,6 +49,26 @@ def convert_fields(fields,heritage=[]):
             convert_fields(df.schema)
             return schema_field
 
+    def extract_unnest(self, df):
+        if isinstance(df, google.cloud.bigquery.table.Table) or isinstance(df, google.cloud.bigquery.table.RowIterator):
+            unnest = []
+            heritage = []
+
+            def convert_fields(fields, heritage=[]):
+                for k in fields:
+                    if k.fields != () and k.is_nullable == False:
+                        heritage.append(k.name)
+                        unnest.append(".".join(heritage))
+                        convert_fields(k.fields, heritage)
+                        heritage = []
```

Comment:
looks a bit weird (or hard to get) logic to me:

```py
heritage.append(k.name)
...
heritage = [] 
# what if original heritage value as passed to `convert_fields(fields, heritage=[])` was not empty [] ?
# => we'd loose it!
# i think you need an extra variable or so, like `current_field_heritage`
```

also if i remember well the default value `heritage=[]` in `convert_fields(fields, heritage=[])`  is mutable and is shared globally among all function calls, so if you modify it inside fn, when you call the fn with default again it will use the old messed up value! so people use this pattern:
```python
def convert_fields(fields, heritage=None):
  if heritage is None:
    heritage = []
  ... your code ...
```

see it yourself!

```py
>>> def f(a=[]):
...   print('a orig:' + str(a))
...   a.append('a')
...   print('a new:' + str(a))
...
>>> f()
a orig:[]
a new:['a']
>>> f()
a orig:['a']
a new:['a', 'a']
>>> f()
a orig:['a', 'a']
a new:['a', 'a', 'a']
```